lib/internal/util.js

```diff
@@ -187,3 +187,18 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+exports.getConstructorOf = function getConstructorOf(obj) {
+  while (obj) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, 'constructor');
```

Comment:
@ljharb, the URL change that added `hasOwnProperty` has not been released yet. In fact, the WHATWG URL API is still in experimental phase, and so compatibility on the URL side is not an issue.

@jseijas the URL constructor is not using this function, only the custom inspection function (which is not performance-sensitive at all) is using it.